autoload/ghpr_blame/slug.vim

```diff
@@ -0,0 +1,69 @@
+let s:H = ghpr_blame#vital().import('Web.HTTP')
+
+let s:SLUG = {}
+function! ghpr_blame#slug#from_url(url) abort
+    let slug = s:SLUG
+    let m = matchlist(a:url, '\v^git\@([^:]+):([^/]+/[^/]{-})%(\.git)?\n*$')
+    if empty(m)
+        let m = matchlist(a:url, '\v^%(git|https|ssh)://%([^@/]+\@)?([^/]+)/([^/]+/[^/]{-})%(\.git)?\n*$')
+    endif
+    if empty(m)
+      let slug.is_valid = v:false
+      return slug
```

Comment:
Invalid slug is an error case. So rather than field of object, I think it should throws an exception. It would be more explicit.

Reply:
Got it. I will throw errors and catch in the app above.